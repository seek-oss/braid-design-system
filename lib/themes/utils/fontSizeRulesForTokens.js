import basekick from 'basekick';
import merge from 'lodash/merge';
import createResponsiveRules from './createResponsiveRules';

export default ({ tokens }) => {
  const size = (fontSize, rows) =>
    basekick({
      baseFontSize: 1,
      typeSizeModifier: fontSize,
      typeRowSpan: rows,
      gridRowHeight: tokens.rowHeight,
      descenderHeightScale: tokens.descenderHeightScale
    });

  const rules = [];
  Object.keys(tokens.text).forEach(typeName => {
    const type = tokens.text[typeName];

    const mobileRules = size(type.mobile.size, type.mobile.rows);
    const desktopRules = size(type.desktop.size, type.desktop.rows);

    rules.push(
      createResponsiveRules({
        tokens,
        selector: `.${typeName}`,
        mobileRules,
        desktopRules
      })
    );

    if (typeName === 'standard') {
      const interactionHeight = tokens.interactionRows * tokens.rowHeight;
      const mobileInteractionPadding = `${(interactionHeight -
        type.mobile.rows * tokens.rowHeight) /
        2}px`;
      const desktopInteractionPadding = `${(interactionHeight -
        type.desktop.rows * tokens.rowHeight) /
        2}px`;

      rules.push(
        createResponsiveRules({
          tokens,
          selector: '.interaction',
          mobileRules: {
            ...mobileRules,
            paddingTop: mobileInteractionPadding,
            paddingBottom: mobileInteractionPadding
          },
          desktopRules: {
            ...mobileRules,
            paddingTop: desktopInteractionPadding,
            paddingBottom: desktopInteractionPadding
          }
        })
      );
    }
  });

  return merge({}, ...rules);
};
