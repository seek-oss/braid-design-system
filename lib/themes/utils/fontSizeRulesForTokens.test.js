import fontSizeRulesForTokens from './fontSizeRulesForTokens';
import forEachTheme from '../../test/utils/forEachTheme';

describe('fontSizeRulesForTokens', () => {
  forEachTheme(({ theme }) => {
    test('Rules', () => {
      expect(
        fontSizeRulesForTokens({ tokens: theme.tokens })
      ).toMatchSnapshot();
    });
  });
});
