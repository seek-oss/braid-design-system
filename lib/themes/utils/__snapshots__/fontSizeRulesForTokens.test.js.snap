// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontSizeRulesForTokens Theme: jobStreet Rules 1`] = `
Object {
  ".interaction": Object {
    "fontSize": "14px",
    "lineHeight": "20px",
    "paddingBottom": "12px",
    "paddingTop": "12px",
    "transform": "translateY(0.3442857142857143em)",
  },
  ".large": Object {
    "fontSize": "18px",
    "lineHeight": "24px",
    "transform": "translateY(0.29666666666666663em)",
  },
  ".standard": Object {
    "fontSize": "14px",
    "lineHeight": "20px",
    "transform": "translateY(0.3442857142857143em)",
  },
  "@media screen and (min-width: 768px)": Object {
    ".large": Object {
      "fontSize": "20px",
      "lineHeight": "28px",
      "transform": "translateY(0.32999999999999996em)",
    },
  },
}
`;

exports[`fontSizeRulesForTokens Theme: seekAnz Rules 1`] = `
Object {
  ".interaction": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "paddingBottom": "12px",
    "paddingTop": "12px",
    "transform": "translateY(0.41000000000000003em)",
  },
  ".large": Object {
    "fontSize": "18px",
    "lineHeight": "24px",
    "transform": "translateY(0.32666666666666666em)",
  },
  ".standard": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "transform": "translateY(0.41000000000000003em)",
  },
}
`;

exports[`fontSizeRulesForTokens Theme: seekAsia Rules 1`] = `
Object {
  ".interaction": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "paddingBottom": "10px",
    "paddingTop": "10px",
    "transform": "translateY(0.38em)",
  },
  ".large": Object {
    "fontSize": "18px",
    "lineHeight": "24px",
    "transform": "translateY(0.29666666666666663em)",
  },
  ".standard": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "transform": "translateY(0.38em)",
  },
  "@media screen and (min-width: 768px)": Object {
    ".large": Object {
      "fontSize": "20px",
      "lineHeight": "28px",
      "transform": "translateY(0.32999999999999996em)",
    },
  },
}
`;

exports[`fontSizeRulesForTokens Theme: wireframe Rules 1`] = `
Object {
  ".interaction": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "paddingBottom": "12px",
    "paddingTop": "12px",
    "transform": "translateY(0.41000000000000003em)",
  },
  ".large": Object {
    "fontSize": "18px",
    "lineHeight": "24px",
    "transform": "translateY(0.32666666666666666em)",
  },
  ".standard": Object {
    "fontSize": "16px",
    "lineHeight": "24px",
    "transform": "translateY(0.41000000000000003em)",
  },
}
`;
